lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -3224,6 +3229,27 @@ static bool diagnoseTypedThrowsAvailability(
       ReferenceDC);
 }
 
+static void diagnoseNonRuntimeProtocol(SourceLoc Loc, TypeDecl *D) {
+  if (auto *ClangD = D->getClangDecl()){
+    auto *proto = dyn_cast<clang::ObjCProtocolDecl>(ClangD);
+    if (!proto || !proto->isNonRuntimeProtocol())
+      return;
+    auto *clangImporter = static_cast<ClangImporter *>(D->getASTContext().getClangModuleLoader());
+    assert(clangImporter && "Must have a clang importer");
+
+
+    auto &Diags = D->getASTContext().Diags;
+    Diags.diagnose(Loc, diag::non_runtime_objc_protocol_metadata_not_available,
+                   D->getNameStr());
+    Diags
+        .diagnose(clangImporter->importSourceLocation(proto->getLocation()),
```

Comment:
Do we need to go to the `ClangDecl` from `D`? I wonder if we actually construct the Swift declaration with the correct ObjC header locations. If that is not the case, I wonder if we should fix that instead of going through the `ClangDecl` here.